android/app/src/main/java/org/uwaterloo/subletr/pages/watcardverification/WatcardVerificationPageView.kt

```diff
@@ -0,0 +1,254 @@
+import org.uwaterloo.subletr.theme.subletrPink
+
+private const val ELEMENT_WIDTH = 0.75f
+@Composable
+fun WatcardVerificationPageView(
+	modifier : Modifier = Modifier,
+	viewModel : WatcardVerificationPageViewModel = hiltViewModel(),
+	uiState: WatcardVerificationUiState = viewModel.uiStateStream.subscribeAsState(
+		WatcardVerificationUiState.Loading
+	).value
+) {
+
+	if (uiState is WatcardVerificationUiState.Loading) {
+		Column(
+			modifier = modifier
+				.fillMaxSize(1.0f)
+				.imePadding(),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			CircularProgressIndicator()
+		}
+	} else if (uiState is WatcardVerificationUiState.Loaded) {
+		Column(
+			modifier = modifier.fillMaxSize(1.0f),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			Spacer(modifier = modifier.weight(8.0f))
+			
+			Text(
+				text = stringResource(id = R.string.watcard_verification),
+				style = MaterialTheme.typography.titleMedium,
+			)
+
+			Column(modifier = modifier
+				.fillMaxWidth(1.0f)
+				.padding(horizontal = dimensionResource(id = R.dimen.l))
+			) {
+				Text(
+					text = stringResource(id = R.string.watcard_verification_txt1),
+					color = secondaryTextColor,
+				)
+				Text(
+					text = stringResource(id = R.string.watcard_verification_txt2),
+					color = secondaryTextColor,
+				)
+			}
+
+			setContent(
+				viewModel = viewModel,
+				watcard = uiState.watcard
+			)
+
+			SetUploadButton(
+				viewModel,
+				uiState.watcard
+			)
+
+			Spacer(modifier = modifier.weight(0.5f))
+
+			setCancelButton(
+				uiState.watcard,
+				uiState.verified,
+			)
+
+			Spacer(modifier = modifier.weight(8.0f))
+		}
+	}
+}
+
+@Composable
+fun setContent(viewModel: WatcardVerificationPageViewModel, watcard : String?) {
+	Box(
+		modifier = Modifier
+			.padding(dimensionResource(id = R.dimen.l))
+			.fillMaxWidth()
+			.height(dimensionResource(id = R.dimen.xxxxxxxxxl))
+			.border(
+				width = dimensionResource(id = R.dimen.xxxs),
+				color = Color.Gray,
+				shape = RoundedCornerShape(dimensionResource(id = R.dimen.zero)),
+			)
+			.background(Color.Transparent)
+			.clickable {
+				viewModel.updateUiState(
+					WatcardVerificationUiState.Loaded(
+						watcard = "uploaded",
+						verified = false
+					)
+				)},
+		contentAlignment = Alignment.Center,
+	) {
+		if (watcard == null) {
+			Column(
+				horizontalAlignment = Alignment.CenterHorizontally,
+				verticalArrangement = Arrangement.Center,
+			) {
+				Text(
+					stringResource(id = R.string.plus),
+					style = MaterialTheme.typography.titleSmall,
+				)
+				Text(
+					stringResource(id = R.string.add_watcard),
+					style = MaterialTheme.typography.bodyMedium,
+				)
+			}
+		} else {
+			// TODO: show picture
+		}
+	}
+}
+
+@Composable
+fun SetUploadButton(
+	viewModel: WatcardVerificationPageViewModel,
+	watcard: String?,
+) {
+	Button(
```

Comment:
you should use `PrimaryButton` here

Reply:
whats the point of the submit button if it changes right after the image is uploaded? I made the upload box clickable so the user clicks that to upload

Reply:
yeah thats basically it, we don't want to send the request to the server as soon as they select an image. Also _ideally_ the user can edit/change the image after they select one (and before they submit) but thats just a nice to have.